irksome/getForm.py

```diff
@@ -1,18 +1,18 @@
 import numpy
-from firedrake import (TestFunction, Function, Constant,
-                       split, DirichletBC, interpolate, project)
+from firedrake import (Constant, DirichletBC, Function, TestFunction,
+                       interpolate, project, split)
 from firedrake.dmhooks import push_parent
 from ufl import diff
 from ufl.algorithms import expand_derivatives
-from ufl.classes import Zero
-from .deriv import TimeDerivative  # , apply_time_derivatives
-from ufl.constantvalue import as_ufl
-from ufl.corealg.multifunction import MultiFunction
 from ufl.algorithms.analysis import has_exact_type
-from ufl.classes import CoefficientDerivative
 from ufl.algorithms.map_integrands import map_integrand_dags
+from ufl.classes import CoefficientDerivative, Zero
+from ufl.constantvalue import as_ufl
+from ufl.corealg.multifunction import MultiFunction
 from ufl.log import error
 
+from .deriv import TimeDerivative  # , apply_time_derivatives
+
 
 class MyReplacer(MultiFunction):
```

Comment:
FWIW, this code is _exactly the same_ as the `replace` functionality in UFL, so could be removed.

Reply:
Last year, we had discussed this -- I had thought I was doing post-order while UFL was doing pre-order traversal.  I was wanting to make sure `Dt(u)` got replaced by itself without `u` first being handled.

Reply:
Ah, my bad, missed that.